lib/bloc_record/collection.rb

```diff
@@ -0,0 +1,32 @@
+module BlocRecord
+  class Collection < Array
+    def update_all(updates)
+      ids = self.map(&:id)
+      self.any? ? self.first.class.update(ids, updates) : false
+    end
+
+    def take(num=1)
+      taken_records = Collection.new
+      if num > 1
+        for i in (0..num) do
+          taken_records << self[i]
+        end
+      else
+        taken_records << self.first
+      end
+      taken_records
+    end
+
+    def where(*args)
+      where_records = Collection.new
+      # assuming the collection Object responds to Object.id syntax
+      # assuming args is a Hash - for string would need regex?
+      self.each do |record|
+        if record.args.first.key == args.first[args.first.key]
```

Comment:
I do not think `record.args` exists.

In this version of your method, you assume `args` is a `Hash`. Since you use the _splat_ operator, it will always be an `Array` containing each argument passed. The _splat_ was necessary for your original implementation, but in this case you want `args` to be the `Hash` passed in, I presume.

The hash can have many key-value pairs - all of which need to be true on each model. This one check is not going to be enough to handle that.